### ISSUE TITLE
Simplify member access inside nameof() may change program behavior

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#21449]
**Version Used**: 15.2, 15.3 Preview 7

:link: Originally reported by [Ville Nurmi](https://developercommunity.visualstudio.com/users/30560/bb74d343-72ed-4263-ad0c-1777ab9c672e.html) on [Developer Community](https://developercommunity.visualstudio.com/content/problem/91761/simplify-member-access-inside-nameof-may-change-pr.html).

**Steps to Reproduce**:

```csharp
using System;
using Foo = SimplifyInsideNameof.Program;

namespace SimplifyInsideNameof
{
  class Program
  {
    static void Main(string[] args)
    {
      Console.WriteLine(nameof(SimplifyInsideNameof.Program)); // <-- IDE0002 wants to put Foo here
    }
  }
}
```

**Expected Behavior**:

IDE0002 is not reported for the use of `nameof(SimplifyInsideNameof.Program)`.

**Actual Behavior**:

IDE0002 is reported. The code fix substitutes `nameof(Foo)`, which changes the program behavior.
